libarchive/ffi.py

```diff
@@ -240,5 +243,7 @@ def ffi(name, argtypes, restype, errcheck=None):
     c_int, check_int)
 ffi('write_finish_entry', [c_archive_p], c_int, check_int)
 
+ffi('write_fail', [c_archive_p], c_int, check_int)
+
```

Comment:
Then shouldn't the context managers call `write_fail` when an exception is raised?